Dependency syntax in EiffelArtifactCreatedEvent

d-stahl-ericsson opened the pull request:
Proposal for dependency declaration syntax in EiffelArtifactCreatedEvent, as described in #10. This is a very open issue, and I'm creating the pull request mostly to solicit feedback.

The added documentation page should give a decent idea of the reasoning and why it's defined as it is.

Please read, reflect and discuss.

rogpers-ericsson commented:
Need to digest this a bit more but some initial comments:

The descriptive documentation contradicts itself, or is at least a bit confusing. In the beginning there is mention of build and run-time dependencies (as different things). At the end it is stated that there is no (syntactical) different between those.

On a more philosophical note...

We have worked to get rid of using GAVs for referencing artifacts (and baselines) in BDEs and instead use the corresponding eventId. This has been for consistency - there can only be one way to reference artifacts (and baselines). This makes sense.

Now, with the improved handling of dependencies we have brought back GAVs as "keys" for dependencies. This is, I guess, the simplest way of handling "external" dependencies and version ranges.

This is not a problem per se: compositions are always done with specific versions while dependencies may be more loose and support ranges.

We need to be very pedagogical when explaining this. The question "why can I not use a GAV for defining my composition (when I use it for dependencies)?" will come.

d-stahl-ericsson commented:
Oops. As you can tell I changed my mind half way through writing the text, and clearly forgot to cover my tracks. Fixed.

Also added some reasoning on why GAVs are the right choice in this case, but not for trace links.
